test/integration/narrow/routes/vet/email.test.js

```diff
@@ -110,6 +116,32 @@ describe('Vet, enter email name test', () => {
       expect(sendMagicLinkEmail.sendVetMagicLinkEmail).toHaveBeenCalledWith(res.request, email.trim(), signupData)
     })
 
+    test.each([
+      { email: validEmail },
+      { email: `  ${validEmail}  ` }
+    ])('returns 302 when payload is valid, sends email with test token and stores email in session (email = "$email")', async ({ email }) => {
+      config.testToken = '1234567890'
+      users.getByEmail.mockResolvedValue({ email: validEmail, isTest: 'yes' })
+
+      const signupData = {}
+      session.getVetSignup.mockReturnValueOnce(signupData)
+      const crumb = await getCrumbs(global.__SERVER__)
+      const options = {
+        headers: { cookie: `crumb=${crumb}` },
+        method: 'POST',
+        payload: { crumb, email },
+        url
+      }
+
+      const res = await global.__SERVER__.inject(options)
+
+      expect(res.statusCode).toBe(302)
+      expect(res.headers.location).toEqual('/vet/check-email')
+      expect(session.setVetSignup).toHaveBeenCalledTimes(1)
+      expect(session.setVetSignup).toHaveBeenCalledWith(res.request, emailKey, email.trim())
+      expect(sendMagicLinkEmail.sendVetMagicLinkEmail).toHaveBeenCalledTimes(1)
+      expect(sendMagicLinkEmail.sendVetMagicLinkEmail).toHaveBeenCalledWith(res.request, email.trim(), signupData)
```

Comment:
I would have expected to see an expect testing that the token was used in this test.

Either to cover the magic link cache was set correctly with the value of the test token here as that would cover lines 12 & 13 in `send-magic-link-email` i.e.
```js
await magiclinkCache.set(email, tokens)
await magiclinkCache.set(token, { email, redirectTo, userType, data })
```

Or and potentially also, to check the call made to `sendEmail` to validate the token was used.